cinema/tests/test_movie_view_set.py

```diff
@@ -0,0 +1,201 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from rest_framework import status
+from rest_framework.reverse import reverse
+from rest_framework.test import APIClient
+
+from cinema.models import Movie, Genre, Actor
+from cinema.serializers import MovieSerializer, MovieListSerializer, MovieDetailSerializer
+
+MOVIE_URL = reverse("cinema:movie-list")
+
+def detail_url(movie_id):
+    return reverse("cinema:movie-detail", args=[movie_id])
+
+def sample_movie(**params) -> Movie:
+    defaults = {
+        "title": "The Great Brick",
+        "description": "The Great Brick fall on superman head",
+        "duration": 121
+    }
+    defaults.update(params)
+    return Movie.objects.create(**defaults)
+
+
+class TestUnauthenticatedUser(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class TestAuthenticatedUser(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="user1@cinema.com",
+            password="2wsxcvfr4",
+        )
+        self.client.force_authenticate(self.user)
```

Comment:
This is a code quality issue. In the `setUp` method, you're creating a user and authenticating the client with this user. This is a good practice as it allows you to reuse this setup in multiple test methods.